controllers/sriovoperatorconfig_controller.go

```diff
@@ -99,6 +100,25 @@ func (r *SriovOperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// Check if we only have one node
+	nodeList := &corev1.NodeList{}
+	err = r.List(context.TODO(), nodeList)
+	if err != nil {
+		logger.Error(err, "Failed to list nodes")
+		return reconcile.Result{}, err
+	}
+
+	// Update the disableDrain field if needed
+	if len(nodeList.Items) == 1 && !defaultConfig.Spec.DisableDrain {
+		defaultConfig.Spec.DisableDrain = true
```

Comment:
Please, add logging here. We need to let users know that configuration is changed

Reply:
this is a one way switch, what if the cluster scales down to one then scales up to more that 1 node ?

Reply:
> This function is only if the config doesn't exist this means on install time or if the user for some reason removes this file.

This is not under the if block at L80, so it will be executed every time Reconcile is called.
maybe user put some configuration, then cluster scaled down (to 1) and operator restarted. this logic would then change the user configuration. later on cluster scaled up and we ended up with drain disabled.

So Here are my thoughts:
1. we document that in case of a single Node cluster, the user should explicitly set DisableDrain 
2. we do this logic only when creating the default object (within if statement block in L80)

Reply:
I also prefer 2 changing the code as requested